From 531c94238c92d2cb3bee420203c13edc23f442f5 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:13 -0400
Subject: [fs] nfsd: fix return of nfs4_acl_write_who

Message-id: <1410898849-22126-10-git-send-email-bfields@redhat.com>
Patchwork-id: 95694
O-Subject: [Patch RHEL7 009/165] nfsd: fix return of nfs4_acl_write_who
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

AFAICT, the only way to hit this error is to pass this function a bogus
"who" value. In that case, we probably don't want to return -1 as that
could get sent back to the client. Turn this into nfserr_serverfault,
which is a more appropriate error for a server bug like this.

Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: f7ce5d284253db9760fc1c3a96b66ec2d9abf0ab
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4acl.c b/fs/nfsd/nfs4acl.c
index 7653b4a..69bca1f 100644
--- a/fs/nfsd/nfs4acl.c
+++ b/fs/nfsd/nfs4acl.c
@@ -869,7 +869,7 @@ __be32 nfs4_acl_write_who(struct xdr_stream *xdr, int who)
   return 0;
  }
  WARN_ON_ONCE(1);
- return -1;
+ return nfserr_serverfault;
 }
 
 EXPORT_SYMBOL(nfs4_acl_new);
-- 
1.7.1